src/components/SecurityShell/SecurityShell.js

```diff
@@ -0,0 +1,264 @@
+import theme from '../../globals/theme';
+
+import Button from '../Button';
+import HeaderNotification from '../SecurityHeader/HeaderNotification';
+import Icon from '../Icon';
+import IconButton from '../IconButton';
+import Link from '../Link';
+import ScrollGradient from '../ScrollGradient';
+
+const namespace = getComponentNamespace('header');
+const shellNamespace = getComponentNamespace('shell');
+const securityShellNamespace = getComponentNamespace('security-shell');
+
+const SecurityShell = ({ children, ...other }) => (
+  <div className={`${shellNamespace} ${securityShellNamespace}`} {...other}>
+    {children}
+  </div>
+);
+
+const SecurityShellHeader = ({ children, ...other }) => (
+  <div className={`${namespace}__container ${securityShellNamespace}__header`}>
+    <header className={namespace} role="banner" {...other}>
+      {children}
+    </header>
+  </div>
+);
+
+const SecurityShellHeaderAction = ({
+  activeAction,
+  children,
+  id,
+  popover,
+  setActiveAction,
+  ...other
+}) => {
+  const isActive = popover && activeAction === id;
+  const ref = useRef(null);
+
+  const onClick = () => {
+    if (isActive) {
+      setActiveAction(null);
+    } else {
+      setActiveAction(id);
+
+      ref.current.focus();
+    }
+  };
+
+  return (
+    <>
+      {cloneElement(children, {
+        onClick,
+        onFocus: () => console.log('onFocus'),
+        ref,
+      })}
+
+      {isActive && (
+        <div
+          className={`${namespace}__popover ${securityShellNamespace}__popover`}
+          {...other}
+        >
+          {popover}
+        </div>
+      )}
+    </>
+  );
+};
+
+const SecurityShellHeaderActions = ({ children, ...other }) => {
+  const [activeAction, setActiveAction] = useState(null);
+
+  return (
+    <div
+      className={`${namespace}__group ${securityShellNamespace}__group`}
+      {...other}
+    >
+      {Children.map(children, (child, index) =>
+        cloneElement(child, {
+          activeAction,
+          id: `${securityShellNamespace}__header__action--${index}`,
+          setActiveAction,
+        })
+      )}
+    </div>
+  );
+};
+
+const SecurityShellHeaderPopoverContent = ({ children, ...other }) => (
+  <div className={`${namespace}__popover__content`} {...other}>
+    {children}
+  </div>
+);
+
+const SecurityShellHeaderPopoverFooter = ({ children, ...other }) => (
+  <footer className={`${namespace}__popover__footer`} {...other}>
+    {children}
+  </footer>
+);
+
+const SecurityShellHeaderPopoverHeader = ({ children, ...other }) => (
+  <section className={`${namespace}__popover__header`} {...other}>
+    <span className={`${namespace}__popover__header__title`}>{children}</span>
+  </section>
+);
+
+const SecurityShellHeaderPopoverNotification = ({ ...other }) => (
+  <li className={`${namespace}__popover__list-item`}>
+    <HeaderNotification {...other} />
+  </li>
+);
+
+const SecurityShellHeaderPopoverNotifications = ({
+  ariaLabel,
```

Comment:
Since this component is more than the close button that will have this aria-label, I wonder if shouldn't call it `closeButtonAriaLabel` to be clearer.